src/app/src/main/kotlin/org/vaccineimpact/orderlyweb/db/AuthorizationRepository.kt

```diff
@@ -140,6 +141,37 @@ class OrderlyAuthorizationRepository : AuthorizationRepository
 
     }
 
+    override fun getReportReaders(reportName: String): Map<User, Scope>
+    {
+        //Returns all users which can read the report, along with their report read scope (global or report-specific)
+        JooqContext().use {
+            val result = it.dsl.select(ORDERLYWEB_USER.USERNAME, ORDERLYWEB_USER.DISPLAY_NAME, ORDERLYWEB_USER.EMAIL,
+                    ORDERLYWEB_USER.USER_SOURCE, ORDERLYWEB_USER.LAST_LOGGED_IN,
+                    ORDERLYWEB_USER_GROUP_GLOBAL_PERMISSION.ID, ORDERLYWEB_USER_GROUP_REPORT_PERMISSION.ID)
+                    .from(ORDERLYWEB_USER)
```

Comment:
Is there any way we can make this query more comprehensible? Ideas: use the `joinPath` util where joins are unambiguous to reduce verbosity, turn abstract permissions for a user into a view (does sqlite support views?), or factor out the abstract permissions temp table from `getAllPermissionsForGroup` below and re-use it here.

Reply:
Could also be worth revisiting the db structure. Am somewhat regretting not going with just having report/version/global permissions table rather than the shared abstract permissions table, given how unwieldy the queries become. Though I do think if we can define views that would help.

Reply:
Sqlite does claim to support views so I'll give that a go.